src/Accounts/Accounts/Rest/InvokeAzRestMethodCommand.cs

```diff
@@ -91,6 +95,17 @@ public class InvokeAzRestMethodCommand : AzureRMCmdlet
         [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Waits for the long-running operation completes before returning the result.")]
+        public SwitchParameter WaitForCompletion { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Specifies the polling header (to fetch from) for long-running operation status.")]
+        [ValidateSet("AzureAsyncLocation", "Location", "OriginalUri", "Operation-Location", IgnoreCase = true)]
+        public string PollFrom { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Specifies the header for final GET result after the long-running operation completes.")]
+        [ValidateSet("FinalStateVia", "Location", "OriginalUri", "Operation-Location", IgnoreCase = true)]
+        public string FinalResultFrom { get; set; }
+
```

Comment:
[ArgumentCompletions  ](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_argument_completion?view=powershell-7.4#argumentcompletions-attribute)seems more proper than ValidateSet. Considering new accepted values may be added in the future, with ArgumentCompletions, we do not need to release new versions for it.

Reply:
Hi @dolauli, it makes sense but this would also allow the user to input any value for the param right? but there will be a limited values for headers that will be appropriate for the FinalResult. So it will result in getting the HeaderNotFoundException quite often right?

Reply:
I have updated this using ArgumentsCompleter :))